gnosis/safe/safe.py

```diff
@@ -989,17 +994,24 @@ def retrieve_modules(
 
         contract = self.contract
         address = SENTINEL_ADDRESS
-        all_modules: List[str] = []
-        while True:
+        all_modules: List[ChecksumAddress] = []
+
+        for _ in range(max_modules_to_retrieve // pagination):
+            # If we use a `while True` loop a custom coded Safe could get us into an infinite loop
             (modules, address) = contract.functions.getModulesPaginated(
                 address, pagination
             ).call(block_identifier=block_identifier)
 
-            all_modules.extend(modules)
-            if address == SENTINEL_ADDRESS:
+            if not modules or address in (NULL_ADDRESS, SENTINEL_ADDRESS):
```

Comment:
Maybe could be a while loop: 
`while modules and address not in (NULL_ADDRESS, SENTINEL_ADDRESS): `

Reply:
If a person deploys a malicious Safe we could get into an infinite loop, that's what I want to prevent